src/main/java/ui/UI.java

```diff
@@ -174,6 +181,12 @@ private void showMainWindow(String repoId) {
         // Should only be called after panels have been initialized
         ensureSelectedPanelHasFocus();
         initialisePickers();
+        //Startup board testing required only on app launch.
+        boolean shouldBeMainLaunchOrStartupBoardTesting =
+                !TestController.isTestMode() || TestController.shouldTestStartupBoard();
+        if (shouldCreateSampleBoardOnAppLaunch && shouldBeMainLaunchOrStartupBoardTesting){
```

Comment:
I feel like these names are at too low a level of abstraction to help readability. The first comment is unnecessary if we use it as the variable name.

```java
boolean isFirstTimeUser = logic.getStoredRepos().isEmpty();
```

I get that giving a succinct name to the `not test mode || has flag` condition is difficult, so we can just give it a general name, and explain why we need it with a header comment (all tests start with a clean slate and we don't want the sample board then, etc.).

```java
if (isFirstTimeUser && shouldOpenSampleBoard()) {
    // ...
}

boolean shouldOpenSampleBoard() {
    return !TestController.isTestMode() || TestController.shouldTestStartupBoard();
}
```

Exact names can be improved, but the level of abstraction should be similar.